launch/launcher_test.go

```diff
@@ -233,15 +260,197 @@ func testLauncher(t *testing.T, when spec.G, it spec.S) {
 
 				if runtime.GOOS == "windows" {
 					h.AssertEq(t, strings.ToLower(syscallExecArgsColl[0].argv0), `c:\windows\system32\notepad.exe`)
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"notepad", "arg1", "arg with space"})
 				} else {
 					h.AssertEq(t, syscallExecArgsColl[0].argv0, "/bin/sh")
-					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg2"})
+					h.AssertEq(t, syscallExecArgsColl[0].argv, []string{"sh", "arg1", "arg with space"})
 				}
 				h.AssertEq(t, syscallExecArgsColl[0].envv, envList)
 			})
 		})
 
+		when("a start command is NOT marked as direct", func() {
+			when("the providing buildpack has api >= 0.4", func() {
```

Comment:
https://github.com/buildpacks/rfcs/blob/main/text/0045-launcher-arguments.md mentions using `CNB_PLATFORM_API` to toggle the launcher process resolution. I'm assuming that we don't need to inspect the platform API version for the arg handling, but please correct me if I'm wrong!

Also, I see that we have a `when` for buildpack API >= 0.4 but would it be worth also having one for lower API versions?

Reply:
Adding a test case for buildpack API < 0.4.

The platform API will come into the picture in the implementation of https://github.com/buildpacks/lifecycle/issues/356

I intend to do some more `launcher_test.go` cleanup at that point as well.